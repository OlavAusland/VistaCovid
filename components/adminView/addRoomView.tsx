import { useState } from "react";
import React, { ScrollView, TextInput, View, Text, Button, SafeAreaView, Alert } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";
import { Room } from "../../domain/RoomType";
import { addRoom, getRooms } from "../../api/firebaseAPI";
import { NoteData } from "../../domain/RoomType";
import { Roles } from "../../domain/UserType";

export function AddRoom()
{

    const [roomNumber, setRoomNumber] = useState("")

 
    const newroom: Room = {
        id: roomNumber,
        patientId: "",
        roomNumber: roomNumber,
        lastUpdated: Date.now().toString(),
        heartRate: [],
        bloodPressure: [],
        oxygenLevel: [],
        notes: []
     }


     const SuccessfullAlert = () =>
     Alert.alert(
       "Successfully added room with room number: "+{roomNumber},
       "",
       [
         { text: "OK", onPress: () => console.log("OK Pressed", roomNumber) }
       ]
     );

     const UnsuccessfullAlert = () =>
     Alert.alert(
       "Could not add room with room number: ",
       "Room might already exist",
       [
         { text: "OK", onPress: () => console.log("OK Pressed") }
       ]
     );

     const AddRoom = () =>{
         addRoom(newroom).then(() => {SuccessfullAlert()}).catch(() => {UnsuccessfullAlert()})
     }

    return(
        <ScrollView style={adminStyle.addRoomContainer}>
            <SafeAreaView>
                <View style={adminStyle.addRoomHeader}>
                    <Text style={adminStyle.headertext}>Add Room</Text>
                </View>
                <View>
                    <TextInput
                    style={adminStyle.addRoomInput}
                    onChangeText={setRoomNumber}
                    placeholder="room number"/>
                    <View style={adminStyle.addRoomButton}>
                        <Button title='Add Room' onPress={()=>{AddRoom()}}/>
                    </View>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}