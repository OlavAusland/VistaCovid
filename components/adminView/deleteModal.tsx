import { Modal, Pressable, Text, View } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";

type DeleteRoomModal={
    modalVisible: boolean;
    handleRequestClose: Function;
    deleteRoomAndClose: Function;
}

export const DeleteRoomModal = (props: DeleteRoomModal) => {
    
    return(
        <Modal
        animationType="slide"
        statusBarTranslucent={true}
        transparent={true}
        visible={props.modalVisible}
        onRequestClose={()=>props.handleRequestClose()}>

                <View style={adminStyle.centeredView}>
                    <View style={adminStyle.modalView}>
                        <Text style={adminStyle.modalText}>Are you sure you want to delete this room?</Text>
                        
                        <Pressable
                        style={[adminStyle.buttonClose, adminStyle.buttonClose]}
                        onPress={()=> props.deleteRoomAndClose()}
                        >
                            <Text style={adminStyle.textStyle}>Yes, delete room</Text>
                        </Pressable>

                        <Pressable
                        style={[adminStyle.buttonClose, adminStyle.buttonClose]}
                        onPress={()=> props.handleRequestClose()}
                        >
                            <Text style={adminStyle.textStyle}>No, close modal</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
    );
}

