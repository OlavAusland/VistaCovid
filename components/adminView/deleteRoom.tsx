import { ScrollView } from "react-native";


export function DeleteRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}