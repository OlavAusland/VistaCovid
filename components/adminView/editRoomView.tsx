import { ScrollView } from "react-native";


export function EditRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}