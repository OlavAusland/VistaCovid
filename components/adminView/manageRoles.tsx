import { useEffect, useState } from "react";
import React, { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable, FlatList, TouchableOpacity, StyleSheet } from "react-native";
import { SearchBar } from 'react-native-elements';

// navigation
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParameters } from '../../domain/NavigationTypes';
import { adminStyle } from "../../styles/AdminStyles";
import { Room } from "../../domain/RoomType";
import { User } from "../../domain/UserType";
import { deletePatient, getUsers, deleteUserById } from "../../api/firebaseAPI";
import { roomStyle } from "../../styles/RoomStyles";
import { isPropertySignature } from "typescript";
import { ErrorType } from "../../domain/Errortype";
import { Errormodal } from "../ErrorModal";
import Icon from "react-native-vector-icons/Feather";

export function ManageRoles(){

    const navigation = useNavigation<NativeStackNavigationProp<StackParameters>>();

    const [searchWord, setSearchWord] = useState("");
    const [users, setUsers] = useState<User[]>([])
    const [search, setSearch] = useState<string>('');
    const [error, setError] = useState<ErrorType>({errorObject:undefined, errormodalVisible:false});

    const handleRequestClose = () => {
        setError((prev) =>({...prev,errorObject:undefined, errormodalVisible:false}));
    }

    useEffect(() => {
        const getUserData = async () =>{
            await getUsers().then((res) => {
                setUsers(res);
            }).catch((err) => { setError((prev) =>({...prev, errorObject:err, errormodalVisible:true})); });
        };
        getUserData();
    }, []);

    if(error.errormodalVisible){
        return (
            <Errormodal error={error} handleRequestClose={handleRequestClose} />
        )
    }

    return(
        <SafeAreaView style={{flex:1}}>
            <View style={[styles.header, styles.shadow]}>
                <TextInput
                    placeholder="room number"
                    onChangeText={(text) => {setSearch(text)}}
                    style={styles.search}/>
            </View>
            <ScrollView contentContainerStyle={{flex:1, alignItems:'center'}}>
                {users.map((user) => {
                    return(
                        <View  key={'users' + user.id} style={styles.cardContainer}>
                            <View style={styles.card}>
                                <Icon style={{alignSelf:'center'}} name='clipboard' size={80}/>
                                <Text style={{flex:1}}>Patient: {user.id}</Text>
                            </View>
                            <TouchableOpacity onPress={() =>{deleteUserById(user.id)}} style={styles.delete}>
                                <Icon name='trash-2' color={'white'} size={40}/>
                            </TouchableOpacity>
                        </View>
                    );
                })}
            </ScrollView>
        </SafeAreaView>
    );
}

export const styles = StyleSheet.create({
    header:{
        backgroundColor:'white',
        paddingTop:40,
        paddingBottom:10
    },
    delete: {
        backgroundColor:'red', 
        flexBasis:'15%', 
        height:100, 
        justifyContent:'center', 
        alignItems:'center', 
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    cardContainer:{
        width:'90%',
        marginTop:25,
        flexDirection:'row', 
        justifyContent:'space-between', 
        alignItems:'center', 
    },
    shadow:{
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3,},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    card: {
        flexDirection:'row', 
        backgroundColor:'#79CAED', 
        flexBasis:'84%', 
        height:100, 
        justifyContent:'center', 
        shadowColor: "#000", 
        shadowOffset: { width: 0,height: 3,},
        shadowOpacity: 0.27,
        shadowRadius: 4.65, 
        elevation: 6
    },
    add:{
        marginTop:25,
        backgroundColor:'white',
        width:'90%',
        height:75,
        justifyContent:'center',
        alignItems:'center',
        borderRadius:10
    },
    search:{
        height: 40,
        margin: 12,
        borderWidth: 1,
        borderRadius:10,
        padding: 10,
    }
})