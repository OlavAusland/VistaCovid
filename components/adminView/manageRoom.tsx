import { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable } from "react-native";
import { useState } from "react";
import { deleteRoom } from "../../api/firebaseAPI";
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParameters } from "../../domain/NavigationTypes";
import { adminStyle } from "../../styles/AdminStyles";
import  handleDelete from "./deleteModal"

export function ManageRoom()
{
    const navigation = useNavigation<NativeStackNavigationProp<StackParameters>>();

    const [roomNumber, setRoomNumber] = useState('')
    const [delroomNumber, setDelRoomNumber] = useState('')



    return(
        <ScrollView>
            <SafeAreaView>
                <View style={adminStyle.editRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setRoomNumber}/>
                    <View style={adminStyle.manageRoomButtons}>
                        <Button title="Edit room" onPress={()=>navigation.navigate("EditRoom")}/>
                    </View>
                </View>

                <View style={adminStyle.deleteRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setDelRoomNumber}/>
                    <Button title="DeleteRoom" onPress={()=>handleDelete}/>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}