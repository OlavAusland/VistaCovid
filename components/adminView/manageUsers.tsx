
import { SearchBar } from 'react-native-elements';
import { manageUsersStyles } from '../../styles/ManageUsersStyles';


// navigation
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useEffect, useState } from "react";
import { SafeAreaView, ScrollView, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";
import Icon from "react-native-vector-icons/Feather";
import { deleteUserById, getUsers } from "../../api/firebaseAPI";
import { ErrorType } from "../../domain/Errortype";
import { StackParameters } from '../../domain/NavigationTypes';
import { User } from "../../domain/UserType";
import { Errormodal } from "../ErrorModal";


export function ManageUsers(){
    const [users, setUsers] = useState<User[]>([])
    const [error, setError] = useState<ErrorType>({errorObject:undefined, errormodalVisible:false});

    const handleRequestClose = () => {
        setError((prev) =>({...prev,errorObject:undefined, errormodalVisible:false}));
    }

    useEffect(() => {
        const getUserData = async () =>{
            await getUsers().then((res) => {
                setUsers(res);
            }).catch((err) => { setError((prev) =>({...prev, errorObject:err, errormodalVisible:true})); });
        };
        getUserData();
    }, []);

    if(error.errormodalVisible){
        return (
            <Errormodal error={error} handleRequestClose={handleRequestClose} />
        )
    }

    return(
        <SafeAreaView style={{flex:1}}>
            <View style={[manageUsersStyles.header, manageUsersStyles.shadow]}>
                <TextInput
                    placeholder="room number"
                    style={manageUsersStyles.search}/>
            </View>
            <ScrollView contentContainerStyle={{alignItems:'center'}}>
                {users.map((user) => {
                    return(
                        <View  key={'users' + user.id} style={manageUsersStyles.cardContainer}>
                            <View style={manageUsersStyles.card}>
                                <Icon style={{alignSelf:'center'}} name='clipboard' size={80}/>
                                <Text style={{flex:1}}>User: {user.firstName} {user.lastName}</Text>
                                <Text style={{flex:1}}>ID: {user.id}</Text>
                            </View>
                        </View>
                    );
                })}
            </ScrollView>
        </SafeAreaView>
    );
}
