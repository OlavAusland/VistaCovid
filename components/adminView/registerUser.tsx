import { useState } from "react";
import React, { Button, SafeAreaView, ScrollView, Text, TextInput, View } from "react-native";
import { Dropdown } from "react-native-element-dropdown";
import FontAwesome5 from "react-native-vector-icons/FontAwesome5";
import { DropDownType } from '../../domain/DropDownType';
import { Roles, User } from '../../domain/UserType';
import { dropdownStyles } from '../../styles/dropdownStyle';


export function CreateUser(){
    const [user, setUser] = useState<User>({
        email:'',  password:'', firstName:'', 
        lastName:'', role:Roles.NONE, address:undefined,
        phone: undefined, city: undefined, code: undefined, id:''
    });
    
    const [dropdown, setDropdown] = useState<DropDownType>({ 
        open: false, 
        value: "0", 
        items: [
            { label: 'Nurse', value: '1' },
            { label: 'Doctor', value: '2' }],
        label: "" });


    const CreateUser = () =>{
        
    }

    return(
        <ScrollView>
            <SafeAreaView>
                <View>
                    <Text>Add new user</Text>
                </View>
                <View>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, email:text}))} placeholder="Email" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, password:text}))} placeholder="Password" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, confirmedPassword:text}))} placeholder="Confirm Password" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, firstName:text}))} placeholder="First Name" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, lastName:text}))} placeholder="Last Name" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, phone:text}))} placeholder="Phone Number" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, address:text}))} placeholder="Address" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <TextInput onChangeText={text => setUser(prev => ({...prev, city:text}))} placeholder="City" style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                    <Dropdown
                            style={dropdownStyles.dropdown}
                            placeholderStyle={dropdownStyles.placeholderStyle}
                            selectedTextStyle={dropdownStyles.selectedTextStyle}
                            inputSearchStyle={dropdownStyles.inputSearchStyle}
                            iconStyle={dropdownStyles.iconStyle}
                            data={dropdown.items}
                            maxHeight={100}
                            labelField="label"
                            valueField="value"
                            placeholder="Role"
                            value={dropdown.label}
                            onChange={item => {
                                setDropdown(prev => ({ ...prev, label: item.value }))
                                setUser(prev => ({ ...prev, role: item.value }));
                            }}
                            renderLeftIcon={() => (
                                <FontAwesome5 style={dropdownStyles.icon} color="black" name="hospital-user" size={20} />
                            )}
                        />
                </View> 
                <View>
                    <Button title="Create User" onPress={() => {}}/>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}