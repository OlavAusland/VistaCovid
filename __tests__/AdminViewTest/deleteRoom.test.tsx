import { DeleteRoomModal } from "../../components/adminView/deleteModal";
import {render } from '@testing-library/react-native';


test("renders correctly", () => {
    const tree = render(<DeleteRoomModal modalVisible={true} handleRequestClose={jest.fn()} deleteRoomAndClose={jest.fn()}    />).toJSON();
    expect(tree).toMatchSnapshot();
});