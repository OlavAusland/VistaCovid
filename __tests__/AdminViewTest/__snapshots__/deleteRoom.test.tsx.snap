// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 5,
          "margin": 20,
          "padding": 35,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 15,
            "textAlign": "center",
          }
        }
      >
        Are you sure you want to delete this room?
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#2196F3",
            },
            Object {
              "backgroundColor": "#2196F3",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Yes, delete room
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#2196F3",
            },
            Object {
              "backgroundColor": "#2196F3",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          No, close modal
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
